### PR TITLE
Bump openssh to 0.8, tokio to 1.0, rand to 0.8, rusoto to 0.46 and ubuntu-ami to 0.2

### PR DESCRIPTION
**Pull request by vitorenesduarte**

### CODE REVIEW

**Comment by vitorenesduarte** [comment]
The current changes are insufficient. I was able to make it work (in a different branch) with a version of openssh that also uses tokio 1.0: https://github.com/vitorenesduarte/tsunami/commit/49e4ba27f0231e3159bd7bbe6f6a6786ca3a5ed2

I'll update this PR once there's a new version of openssh.

**Comment by jonhoo** [comment]
There's now a new version of openssh! :tada:

**Comment by vitorenesduarte** [comment]
Awesome! This PR should be ready now.

File: Cargo.toml
Comment (jonhoo) [inline Cargo.toml]: I think you mean `0.8`?

File: Cargo.toml
Comment (jonhoo) [inline Cargo.toml]: Let's bump `rand` too while we're at it. Or in a separate PR if you prefer.

**Comment by jonhoo** [comment]
I think we still need to hold off a bit here. Even after the bumps here, `cargo tree` tells me that we have some outstanding dependencies on old `tokio`:
```console
$ cargo tree -i tokio:0.2.24
tokio v0.2.24
├── h2 v0.2.7
│   └── hyper v0.13.9
│       ├── hyper-tls v0.4.3
│       │   ├── reqwest v0.10.10
│       │   │   └── ubuntu-ami v0.1.2
│       │   │       └── tsunami v0.11.0-beta.10 (/home/jon/dev/minor/tsunami)
│       │   └── rusoto_core v0.45.0
│       │       ├── rusoto_ec2 v0.45.0
│       │       │   └── tsunami v0.11.0-beta.10 (/home/jon/dev/minor/tsunami)
│       │       ├── rusoto_sts v0.45.0
│       │       │   [dev-dependencies]
│       │       │   └── tsunami v0.11.0-beta.10 (/home/jon/dev/minor/tsunami)
│       │       └── tsunami v0.11.0-beta.10 (/home/jon/dev/minor/tsunami)
│       ├── reqwest v0.10.10 (*)
│       ├── rusoto_core v0.45.0 (*)
│       ├── rusoto_credential v0.45.0
│       │   ├── rusoto_core v0.45.0 (*)
│       │   └── rusoto_signature v0.45.0
│       │       └── rusoto_core v0.45.0 (*)
│       └── rusoto_signature v0.45.0 (*)
```
Specifically, through `ubuntu-ami` which [uses `reqwest 0.10`](https://github.com/akshayknarayan/ubuntu-ami/blob/668af700e40ef3e739d20613346c9a289164ad82/Cargo.toml#L24): https://github.com/seanmonstar/reqwest/issues/1123), and through `rusoto 0.45`: https://github.com/rusoto/rusoto/issues/1845. We should probably wait for those, or the bump won't do much (and probably won't even work!)

File: Cargo.toml
Comment (vitorenesduarte) [reply Cargo.toml]: Yeah. (Just noticed it and force-pushed, sorry.)

File: Cargo.toml
Comment (vitorenesduarte) [reply Cargo.toml]: Done.

**Comment by vitorenesduarte** [comment]
Interesting. It was working for me, but probably only because I was using the baremetal provider. I'll keep an eye on those issues.

**Comment by jonhoo** [comment]
See also #115, which might be a way for us to get away from a dependency on `hyper-tls`.

**Comment by vitorenesduarte** [comment]
You forgot the link 😛

**Comment by jonhoo** [comment]
Oh, ignore me, I got projects mixed up in my head when commenting :sweat_smile: I meant https://github.com/jonhoo/fantoccini/pull/115, but that's not relevant to this!

**Comment by vitorenesduarte** [comment]
Once the following PRs are merged and new a version is released, we should be good to go:
- [x] https://github.com/akshayknarayan/ubuntu-ami/pull/1
- [x] https://github.com/rusoto/rusoto/pull/1880

**Comment by akshayknarayan** [comment]
I merged https://github.com/akshayknarayan/ubuntu-ami/pull/1 and released it as v0.2.0: https://crates.io/crates/ubuntu-ami

**Comment by vitorenesduarte** [comment]
Awesome. Thanks @akshayknarayan!

**Comment by benesch** [comment]
I just shipped rusoto v0.46.0!

**Comment by vitorenesduarte** [comment]
Great news! Thank you for your work @benesch.

**Comment by jonhoo** [comment]
The CI failure is because we're still pinning nightly, and a fairly old one at that. You can probably update all instances of `nightly-2020-05-15` to just say `nightly` in `azure-pipelines.yml` now, and we should be good! And then we can do a follow-up PR that moves to running against `stable` and `beta` as well.

**Comment by vitorenesduarte** [comment]
Thanks, it looks like it will pass this time. I'll also push a commit fixing/ignoring the clippy warnings I noticed in CI. Let me know if I should revert it or address those warnings in some other way.

File: src/lib.rs
Comment (vitorenesduarte) [inline src/lib.rs]: I tried to address these warnings but I found the code less readable afterwards.

Comment (jonhoo) [reply src/lib.rs]: Hmm, this lint _should_ be a fairly good one. What did you find problematic about it?
Also, I think we can remove the `allow(clippy::type_complexity)`, since that lint should now be off by default anyway.

Comment (vitorenesduarte) [reply src/lib.rs]: Sorry for the delay. Regarding `allow(clippy::field_reassign_with_default)`, we're getting warnings in:
1. https://github.com/jonhoo/tsunami/blob/cfcb7df8daa360671fb8b185822c9052bd846454/src/providers/aws.rs#L834-L836
2. https://github.com/jonhoo/tsunami/blob/cfcb7df8daa360671fb8b185822c9052bd846454/src/providers/aws.rs#L846-L853
3. https://github.com/jonhoo/tsunami/blob/cfcb7df8daa360671fb8b185822c9052bd846454/src/providers/aws.rs#L1004-L1019
4. https://github.com/jonhoo/tsunami/blob/cfcb7df8daa360671fb8b185822c9052bd846454/src/providers/aws.rs#L1099-L1111
5. https://github.com/jonhoo/tsunami/blob/cfcb7df8daa360671fb8b185822c9052bd846454/src/providers/aws.rs#L1439-L1440
6. https://github.com/jonhoo/tsunami/blob/cfcb7df8daa360671fb8b185822c9052bd846454/src/providers/aws.rs#L1478-L1479
7. https://github.com/jonhoo/tsunami/blob/cfcb7df8daa360671fb8b185822c9052bd846454/src/providers/aws.rs#L1519-L1520
8. https://github.com/jonhoo/tsunami/blob/cfcb7df8daa360671fb8b185822c9052bd846454/src/providers/aws.rs#L1569-L1570
9. https://github.com/jonhoo/tsunami/blob/cfcb7df8daa360671fb8b185822c9052bd846454/src/providers/aws.rs#L1619-L1620

The ones I found awkward were 3 and 4, but probably due to the `self.make_placement` request. I'm now thinking that if that request is performed before the initialization, then these changes would be fine. What do you think?

As for `allow(clippy::type_complexity)`, if we remove it, we get a warning for example in: https://github.com/jonhoo/tsunami/blob/cfcb7df8daa360671fb8b185822c9052bd846454/src/lib.rs#L335-L337

Comment (vitorenesduarte) [reply src/lib.rs]: I went ahead and did it in 9229fd515310660c9bbbe1b2303d1e00d691fe8a.

**Comment by jonhoo** [comment]
Oh, I was getting `type_complexity` (which is still on by default) and `cyclomatic_complexity` (which is now off by default) mixed up!

**Review by jonhoo — APPROVED** [review]

**Comment by jonhoo** [comment]
Awesome, thank you for the work! Will release shortly.

**Comment by jonhoo** [comment]
Released in `0.11.0` :tada:

**Pull request merged**
